cms/models.py

```diff
@@ -766,9 +766,7 @@ def is_internal_or_external_course_page(self):
     @property
     def external_courseware_url(self):
         """Gets the product page type, this is used for sorting product pages."""
-        if self.product and self.product.first_unexpired_run:
-            return self.product.first_unexpired_run.external_marketing_url
-        return ""
+        return getattr(self.product, "marketing_url", "")
```

Comment:
I think `getattr` is redundant here. We can simply do `self.product.marketing_url`. No?

Reply:
The default value when no URL is added should be `""` instead of `None` and this is the existing behaviour and `Learn More` button would keep the user on the same page.

>I think getattr is redundant here. We can simply do self.product.marketing_url. No?

The product could be `None` too and we'll get `NoneType object has no...` this way it's just being done in a single line.

Reply:
Oh yeah, Thanks. My bad!